packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -122,6 +130,20 @@ class Icon extends StatelessWidget {
     );
   }
 
+  /// Wraps the widget with a Semantics widget if [label] is set.
+  Widget wrapWithSemantics(Widget widget) {
```

Comment:
Maybe move the ExcludeSemantics to outside of this method since only one call-side needs it and then simplify to:
```dart
if (label == null)
  return widget;
return new Semantics(label: label, child: widget);
```